crawler/file_processing.py

```diff
@@ -663,60 +717,394 @@ def format_and_filter_rows(self, csvreader: DictReader) -> Any:
+            return None
 
-                if row[FIELD_PLATE_BARCODE]:
-                    row[FIELD_LINE_NUMBER] = line_number  # type: ignore
-                    row[FIELD_FILE_NAME] = self.file_name
-                    row[FIELD_FILE_NAME_DATE] = self.file_name_date()
-                    row[FIELD_CREATED_AT] = import_timestamp
-                    row[FIELD_UPDATED_AT] = import_timestamp
+        # check if the result value matches to expected
+        if not self.row_result_value_valid(modified_row, line_number):
+            return None
 
-                    row_signature = self.get_row_signature(row)
+        # check if the ct channel target values match to expected (if present)
+        if not self.row_channel_target_values_valid(modified_row, line_number):
+            return None
 
-                    if row_signature in seen_rows:
-                        logger.debug(f"Skipping {row_signature}: duplicate")
-                        self.logging_collection.add_error(
-                            "TYPE 5",
-                            f"Duplicated, line: {line_number}, root_sample_id: {row[FIELD_ROOT_SAMPLE_ID]}",
-                        )
-                        continue
-                    seen_rows.add(row_signature)
-                    augmented_data.append(row)
-                else:
-                    missing_data_count += 1
+        # check if the ct channel result values match to expected (if present)
+        if not self.row_channel_result_values_valid(modified_row, line_number):
+            return None
+
+        if not self.row_channel_cq_values_valid(modified_row, line_number):
+            return None
+
+        if not self.row_channel_cq_values_within_range(modified_row, line_number):
+            return None
+
+        if not self.row_positive_result_matches_channel_results(modified_row, line_number):
+            return None
+
+        # check if this row has already been seen in this file
+        row_signature = self.get_row_signature(modified_row)
+
+        if row_signature in seen_rows:
+            logger.debug(f"Skipping {row_signature}: duplicate")
+            self.logging_collection.add_error(
+                "TYPE 5",
+                f"Duplicated, line: {line_number}, root_sample_id: {modified_row[FIELD_ROOT_SAMPLE_ID]}",
+            )
+            return None
+
+        # add file details and import timestamp
+        import_timestamp = self.get_now_timestamp()
+
+        modified_row[FIELD_LINE_NUMBER] = line_number  # type: ignore
+        modified_row[FIELD_FILE_NAME] = self.file_name
+        modified_row[FIELD_FILE_NAME_DATE] = self.file_name_date()
+        modified_row[FIELD_CREATED_AT] = import_timestamp
+        modified_row[FIELD_UPDATED_AT] = import_timestamp
+
+        # store row signature to allow checking for duplicates in follwing rows
+        seen_rows.add(row_signature)
+
+        return modified_row
+
+    def row_result_value_valid(self, row, line_number) -> bool:
```

Comment:
Agreed, we could compact down this validation. 
Might be nice to move it into a separate field_validation.py file too, or similar? See what you think after you've done your DRYing of code.